cmd/kubeapps-apis/server/plugins_test.go

```diff
@@ -240,3 +243,120 @@ func createTestFS(t *testing.T, filenames []string) fstest.MapFS {
+
+func TestExtractToken(t *testing.T) {
+	testCases := []struct {
+		name          string
+		contextKey    string
+		contextValue  string
+		expectedToken string
+		expectedErr   string
+	}{
+		{
+			name:          "Good token",
+			contextKey:    "authorization",
+			contextValue:  "Bearer abc",
+			expectedToken: "abc",
+			expectedErr:   "",
+		},
+		{
+			name:          "Malformed bearer token",
+			contextKey:    "authorization",
+			contextValue:  "Bla",
+			expectedToken: "",
+			expectedErr:   "rpc error: code = Unauthenticated desc = malformed authorization metadata",
+		},
+		{
+			name:          "No metadata/header",
+			contextKey:    "",
+			contextValue:  "",
+			expectedToken: "",
+			expectedErr:   "",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			context := context.Background()
+			context = metadata.NewIncomingContext(context, metadata.New(map[string]string{
+				tc.contextKey: tc.contextValue,
+			}))
+
+			token, err := extractToken(context)
+
+			if tc.expectedErr != "" && err != nil {
+				if got, want := err.Error(), tc.expectedErr; !cmp.Equal(want, got) {
+					t.Errorf("in %s: mismatch (-want +got):\n%s", tc.name, cmp.Diff(want, got))
+				}
+			} else if err != nil {
+				t.Fatalf("in %s: %+v", tc.name, err)
+			}
+
+			if got, want := token, tc.expectedToken; !cmp.Equal(want, got) {
+				t.Errorf("in %s: mismatch (-want +got):\n%s", tc.name, cmp.Diff(want, got))
+			}
+		})
+	}
+}
+
+func TestDynClientGetterForContext(t *testing.T) {
+	testCases := []struct {
+		name         string
+		contextKey   string
+		contextValue string
+		shouldCreate bool
+	}{
+		{
+			name:         "Good token",
+			contextKey:   "authorization",
+			contextValue: "Bearer abc",
+			shouldCreate: true,
+		},
+		{
+			name:         "Malformed bearer token",
+			contextKey:   "authorization",
+			contextValue: "Bla",
+			shouldCreate: false,
+		},
+		{
+			name:         "No metadata/header",
+			contextKey:   "",
+			contextValue: "",
+			shouldCreate: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			context := context.Background()
+			context = metadata.NewIncomingContext(context, metadata.New(map[string]string{
+				tc.contextKey: tc.contextValue,
+			}))
+
+			inClusterConfig := &rest.Config{}
+			serveOpts := ServeOptions{
+				ClustersConfigPath: "/config.yaml",
+				PinnipedProxyURL:   "http://example.com",
+				UnsafeUseDemoSA:    true,
+			}
+			config := kube.ClustersConfig{
+				KubeappsClusterName: "default",
+				PinnipedProxyURL:    serveOpts.PinnipedProxyURL,
+				Clusters: map[string]kube.ClusterConfig{
+					"default": {
+						Name: "default",
+						PinnipedConfig: kube.PinnipedConciergeConfig{
+							Enable: true,
+						},
+						IsKubeappsCluster: true,
+					},
+				},
+			}
+			dynamicInterface, err := dynClientGetterForContextWithConfig(context, inClusterConfig, serveOpts, config)
+
+			if tc.shouldCreate != (dynamicInterface != nil) {
```

Comment:
Exactly, I initally was trying to -use- the dynamic interface, but it was more tricky than expected, so I just checked if the creation was being performed. 

I've added a //TODO there, but I'd rather leave the current test. At least we are checking the closure fn is being called with the proper params so that it doesn't throw any error.